src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -19,49 +20,95 @@
+                .filter(oldSection -> section.getUpStation() == oldSection.getUpStation())
+                .findFirst()
+                .ifPresent(oldSection -> {
+                    sections.add(new Section(oldSection.getLine(), section.getDownStation(), oldSection.getDownStation(), getUpStationDistance(section, oldSection)));
+                    sections.remove(oldSection);
+                });
     }
 
-    private Section isNotContainDownStations(Station upStation) {
-        return this.sections
-                .stream()
-                .filter(section -> isNotEqualsDownStation(section, upStation))
+    private void addDownToDownStation(Section section) {
+        sections.stream()
+                .filter(oldSection -> section.getUpStation() == oldSection.getDownStation())
                 .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException("순환 구간은 없습니다."));
+                .ifPresent(oldSection -> {
+                    sections.add(new Section(oldSection.getLine(), oldSection.getUpStation(), section.getUpStation(), section.getDistance().get()));
+                    sections.remove(oldSection);
+                });
     }
 
-    private boolean isNotEqualsDownStation(Section section, Station upStation) {
-        return !section.getDownStation().getId().equals(upStation.getId());
+    private long getUpStationDistance(Section section, Section oldSection) {
+        long nextDistance = oldSection.getDistance().get() - section.getDistance().get();
+        if (nextDistance < 1) {
+            throw new IllegalArgumentException("기존 역 사이 길이보다 크거나 같으면 등록을 할 수 없음");
+        }
+        return nextDistance;
     }
 
-    private Section getNextSection(Station downStation) {
-        return this.sections
-                .stream()
-                .filter(section -> isNextStation(section, downStation))
+    private Section findStartSection() {
+        return sections.stream()
+                .filter(s -> !isContainsDownStation(s.getUpStation()))
                 .findFirst()
-                .orElse(null);
+                .orElseThrow(RuntimeException::new);
+    }
+
+    private boolean isContainsDownStation(Station station) {
+        boolean flag = false;
+        for (Section section : sections) {
+            flag = section.getDownStation() == station;
+        }
+        return flag;
+    }
+
+    private Optional<Section> findNextSection(Station downStation) {
+        return sections.stream()
+                .filter(section -> section.getUpStation() == downStation)
+                .findFirst();
     }
 
-    private boolean isNextStation(Section section, Station downStation) {
-        return section.getUpStation().getId().equals(downStation.getId());
+    private void validation(Section section) {
+        if (sections.size() == 0) return;
+        int containsCount = getContainsCount(section);
+        stationExistValid(containsCount);
+    }
+
+    private int getContainsCount(Section section) {
```

Comment:
어떤게 포함된 카운트인지, 아래처럼 명확하게 남겨주시면 어떨까요?

```suggestion
    private int getContainsStationCount(Section section) {
```